ancestry/extension/__init__.py

```diff
@@ -35,76 +38,106 @@
+            project.ancestry.add(person_name)
+            await user.message_information(
+                plain(f"Published {person_name.label.localize(DEFAULT_LOCALIZER)}")
+            )
+
+
+class _PublishBart(Job[ProjectContext]):
+    def __init__(self):
+        super().__init__(
+            "ancestry:publish-bart", dependencies={PrivatizeAncestry.id_for()}
+        )
+
+    @override
+    async def do(self, scheduler: Scheduler[ProjectContext], /) -> None:
+        project = scheduler.context.project
+        user = project.app.user
+        await user.message_information(plain("Publishing Bart..."))
+        bart = project.ancestry[Person]["I0000"]
+        netherlands = project.ancestry[Place]["P0052"]
         birth = Event(
             event_type=Birth(),
             date=DateRange(Date(1970, 1, 1), start_is_boundary=True),
             place=netherlands,
             public=True,
         )
         Presence(bart, Subject(), birth)
-        self.project.ancestry.add(birth)
+        project.ancestry.add(birth)
         for presence in bart.presences:
             if isinstance(presence.event.event_type, Conference):
                 presence.public = True
                 presence.event.public = True
 
-    async def _publish_files(self, event: PostLoadAncestryEvent):
-        logger = logging.getLogger("betty")
-        logger.info("Publishing selected files...")
+
+class _PublishFiles(Job[ProjectContext]):
+    def __init__(self):
+        super().__init__(
+            "ancestry:publish-files", dependencies={PrivatizeAncestry.id_for()}
+        )
+
+    @override
+    async def do(self, scheduler: Scheduler[ProjectContext], /) -> None:
+        project = scheduler.context.project
+        user = project.app.user
+        await user.message_information(plain("Publishing selected files..."))
         for file_id in _FILES:
-            file = self.project.ancestry[File][file_id]
+            file = project.ancestry[File][file_id]
             file.public = True
-            logger.info(f"Published {file.label.localize(DEFAULT_LOCALIZER)}")
+            await user.message_information(
+                plain(f"Published {file.label.localize(DEFAULT_LOCALIZER)}")
+            )
+
+
+class Ancestry(NavigationLinkProvider, PostLoader, Extension):
+    @override
+    @classmethod
+    def depends_on(cls) -> set[PluginIdentifier[Extension]]:
+        return {Privatizer}
+
+    @override
+    @classmethod
+    def plugin_id(cls) -> MachineName:
+        return "ancestry"
+
+    @override
+    @classmethod
+    def plugin_label(cls) -> Localizable:
+        return static("Publish people")
+
+    @override
+    @classmethod
+    def plugin_description(cls) -> Localizable:
+        return static("Publishes curated information about selected people.")
+
+    @override
+    async def post_load(self, scheduler: Scheduler[ProjectContext]) -> None:
+        await scheduler.add(
+            _PublishPeople(),
```

Comment:
Merge these into a single job that just calls separate functions to populate the individual bits.